### PR TITLE
Move otelexporter code from its own module to AADAuthenticator

### PR DESCRIPTION
**Pull request by shahzaibj**
Related: https://github.com/AzureAD/ad-accounts-for-android/pull/2675

### CODE REVIEW

File: testapps/testapp/src/main/java/com/microsoft/identity/client/testapp/MainActivity.java [hunk 1]
```diff
@@ -168,15 +172,28 @@ protected void onCreate(Bundle savedInstanceState) {
      * @param applicationContext the application context
      */
     private static synchronized void initOpenTelemetry(@lombok.NonNull final Context applicationContext) {
-        AriaInitializer.initializeAria(applicationContext, BuildConfig.otelAriaToken);
         final Resource resource = Resource.getDefault();
 
-        final AriaSpanExporter ariaSpanExporter = new AriaSpanExporter(
```
Comment (p3dr0rv) [inline testapps/testapp/src/main/java/com/microsoft/identity/client/testapp/MainActivity.java:174]: @shahzaibj could you please explain why this change?

Comment (shahzaibj) [reply testapps/testapp/src/main/java/com/microsoft/identity/client/testapp/MainActivity.java:174]: AriaSpanExporter was being used to upload spans created in msal test app to Aria (These are spans in MSAL layer - not in the Broker). This was done to test span context propagation work I did for Yammer. Now that otelexporter is removed as local dependency I removed this piece of code as well. The same test can be performed by just printing msal side spans to console.

Alternatively we can keep dist dependency (on otelexporter already published) but I didn't see a reason to do that.

**Review by rpdome — APPROVED** [review]

**Review by p3dr0rv — APPROVED** [review]
